snappi/snappicommon.py

```diff
@@ -141,15 +141,30 @@ def parent(self):
         return self._parent
         
     def _get_property(self, name, default_value=None, parent=None, choice=None):
-        if name not in self._properties or self._properties[name] is None:
-            if isinstance(default_value, type) is True:
-                self._properties[name] = default_value(parent=parent, choice=choice)
+        if name in self._properties and self._properties[name] is not None:
+            return self._properties[name]
+        if isinstance(default_value, type) is True:
+            if 'choice' in dir(self):
+                self._properties['choice'] = name
+            self._properties[name] = default_value(parent=parent, choice=choice)
+
+            if '_DEFAULTS' in dir(self._properties[name]) and\
+                'choice' in self._properties[name]._DEFAULTS:
+                getattr(self._properties[name], self._properties[name]._DEFAULTS['choice'])
+        else:
+            if default_value is None and name in self._DEFAULTS:
+                if 'choice' in dir(self):
+                    self._properties['choice'] = name
```

Comment:
same concern regarding what if `name` is not a valid enum in `choice`.

Reply:
will update the branch